src/db.c

```diff
@@ -7850,13 +7838,13 @@ int is_trailpoint_echo(DataRow *p_station) {
 
         if ((p_station->coord_lon == ptr->trail_long_pos)
                 && (p_station->coord_lat == ptr->trail_lat_pos)
-                && (p_station->speed == '\0' || ptr->speed < 0
+                && (p_station->speed[0] == '\0' || ptr->speed < 0
```

Comment:
Are these three changes correct? Looks like p_station points to a DataRow which has "char speed" in it. Might be getting confused with TrackRow which is also attached to DataRow and has speed? Same question of course for course/altitude.

Reply:
I believe these are correct.

DataRow has these elements declared as char arrays, not as char pointers.  The test is intended to test that the string stored in the char array is not empty.  It was instead comparing the pointer to the first element to the null character, instead of the first element to a null character.